src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingProviderConfig.java

```diff
@@ -10,23 +10,11 @@
 import java.util.Optional;
 import org.eclipse.microprofile.config.spi.Converter;
 
-@ConfigMapping(prefix = "jsonapi.embedding.provider")
+@ConfigMapping(prefix = "jsonapi.embedding")
 public interface PropertyBasedEmbeddingProviderConfig {
 
   @Nullable
-  EmbeddingProviderConfig openai();
-
-  @Nullable
-  EmbeddingProviderConfig huggingface();
-
-  @Nullable
-  EmbeddingProviderConfig vertexai();
-
-  @Nullable
-  EmbeddingProviderConfig cohere();
-
-  @Nullable
-  EmbeddingProviderConfig nvidia();
+  Map<String, EmbeddingProviderConfig> providers();
```

Comment:
Why we need to change this?

Reply:
If I understand correctly, we previously hard-coded provider names as fields. It's not convenient when checking supported providers. By changing to this, we can just use the key of the map.